twoliter/embedded/rpm2img

```diff
@@ -186,61 +186,73 @@ INVENTORY_DATA="$(jq --raw-output . <<<"${installed_rpms[@]}")"
+  fi
+
+  # Set the path inside the build container to the kit RPMs and repo.
+  KIT_PATH="${EXTERNAL_KITS_PATH}/${KIT_VENDOR}/${kit}/${ARCH}"
+
+  # Query the kit repo for a single package's Release in order
+  # to extract the commit that built the kit.
+  KIT_GIT_SHA="$(dnf --repofrompath \
+    kit,file://"${KIT_PATH}" \
+    --repo=kit repoquery \
     --queryformat '%{Buildtime} %{Release}' 2>/dev/null | \
     sort -k 1,2 | \
     awk -F '.' 'END {print $--NF}')"
-  if [[ -z "${CORE_KIT_GIT_SHA}" ]]; then
-    echo "Could not find Git sha for bottlerocket-core-kit" >&2
+  if [[ -z "${KIT_GIT_SHA}" ]]; then
+    echo "Could not find Git sha for kit '${kit}'" >&2
     exit 1
   fi
   # If the Git sha contains whitespace, we may have accidentally grabbed multiple
-  if [[ "${CORE_KIT_GIT_SHA}" =~ [[:space:]] ]]; then
-    echo "Extracted invalid Git sha from bottlerocket-core-kit: '${CORE_KIT_GIT_SHA}'" >&2
+  if [[ "${KIT_GIT_SHA}" =~ [[:space:]] ]]; then
+    echo "Extracted invalid Git sha from kit '${kit}': '${KIT_GIT_SHA}'" >&2
     exit 1
   fi
 
-  # Query the bottlerocket-core-kit repo of RPMs for all package names.
-  CORE_KIT_INVENTORY_QUERY="%{NAME}"
+  # Query the kit repo of RPMs for all package names.
+  KIT_INVENTORY_QUERY="%{NAME}"
   # shellcheck disable=SC2312 # Array is validated elsewhere.
-  mapfile -t CORE_KIT_PKGS <<<"$(dnf --repofrompath \
-    core-kit,file://"${CORE_KIT_PATH}" \
-    --repo=core-kit repoquery \
-    --queryformat "${CORE_KIT_INVENTORY_QUERY}")"
-  # Convert the bash array of core kit packages to a JSON array.
-  CORE_KIT_LIST="$(\
+  mapfile -t KIT_PKGS <<<"$(dnf --repofrompath \
+    kit,file://"${KIT_PATH}" \
+    --repo=kit repoquery \
+    --queryformat "${KIT_INVENTORY_QUERY}")"
+  # Convert the bash array of kit packages to a JSON array.
+  KIT_LIST="$(\
     jq \
       --null-input \
       --compact-output \
       '$ARGS.positional // []' \
-      --args "${CORE_KIT_PKGS[@]}")"
+      --args "${KIT_PKGS[@]}")"
 
   # Convert the JSON array to a map of 'bottlerocket-' prefixed names to unprefixed package names
   # for search and replace in the installed application inventory.
   jq \
     --compact-output \
-    'map({ (.|tostring): (.|sub("^bottlerocket-";""))}) | add' <<<"${CORE_KIT_LIST}" \
-    > core-kit-replacements.json
+    'map({ (.|tostring): (.|sub("^bottlerocket-";""))}) | add' <<<"${KIT_LIST}" \
+    > kit-replacements.json
 
-  # For any packages in the installed app inventory that exist in the core kit, replace
-  # the version with the core kit's version, and replace the name with the unprefixed name.
+  # For any packages in the installed app inventory that came from the kit, replace
+  # the Publisher with the kit's vendor and replace the name with the unprefixed name.
   INVENTORY_DATA="$(jq \
-    --argfile replace core-kit-replacements.json \
-     '.Content[].Name |= (if $replace[.] then $replace[.] else . end)' \
+    --argfile replace kit-replacements.json \
```

Comment:
While you're here, can you get rid of this `--argfile` usage? It's apparently been deprecated for a long time and was removed in jq 1.7.